packages/views.py

```diff
@@ -83,53 +120,63 @@ def partner_new_order(request):
         origin_area = request.POST.get('origin_area')
         origin_street = request.POST.get('origin_street')
         origin_street_number = request.POST.get('origin_street_number')
-        origin_address = get_address(origin_city, origin_area, origin_street, origin_street_number)
+        origin_address = get_or_create_address(origin_city, origin_area, origin_street, origin_street_number)
 
         destination_city = request.POST.get('destination_city')
         destination_area = request.POST.get('destination_area')
         destination_street = request.POST.get('destination_street')
         destination_street_number = request.POST.get('destination_street_number')
-        destination_address = get_address(destination_city, destination_area, destination_street, destination_street_number)
+        destination_address = get_or_create_address(destination_city, destination_area, destination_street, destination_street_number)
 
-        rate = float(request.POST.get('rate'))
+        rate = float(request.POST.get('rate').replace('₪', ''))
         phone_number = request.POST.get('phone_number')
+        full_name = request.POST.get('full_name')
 
-        Package.objects.create(origin=origin_address, destination=destination_address, order=order, rate=rate, phone_number=phone_number)
+        Package.objects.create(origin=origin_address, destination=destination_address, order=order, rate=rate,
+                               full_name=full_name, phone_number=phone_number)
 
     context = {
         'packages': order.related_packages(),
         'order': order,
         'partner': partner,
         'rates': partner.rates.split(','),
-        'cities': City.objects.all()
+    }
+    context += string_data_lists_context()

```

Comment:
You can't concatenate two `dict` objects.

Reply:
`.update()`